src/test/java/lambda/part1/exercise/Exercise1.java

```diff
@@ -0,0 +1,120 @@
+package lambda.part1.exercise;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
+import java.util.Comparator;
+import lambda.data.Person;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+public class Exercise1 {
+
+  @Test
+  public void sortPersonsByAgeUsingArraysSortComparator() {
+    Person[] persons = getPersons();
+
+    class AgeComparator implements Comparator<Person> {
+
+      @Override
+      public int compare(Person left, Person right) {
+        return Integer.compare(left.getAge(), right.getAge());
+      }
+    }
+    Arrays.sort(persons, new AgeComparator());
+
+    assertArrayEquals(new Person[]{
+        new Person("Иван", "Мельников", 20),
+        new Person("Николай", "Зимов", 30),
+        new Person("Алексей", "Доренко", 40),
+        new Person("Артем", "Зимов", 45)
+    }, persons);
+  }
+
+  @Test
+  public void sortPersonsByAgeUsingArraysSortAnonymousComparator() {
+    Person[] persons = getPersons();
+
+    Arrays.sort(persons, new Comparator<Person>() {
+      @Override
+      public int compare(Person left, Person right) {
+        return Integer.compare(left.getAge(), right.getAge());
+      }
+    });
+
+    assertArrayEquals(new Person[]{
+        new Person("Иван", "Мельников", 20),
+        new Person("Николай", "Зимов", 30),
+        new Person("Алексей", "Доренко", 40),
+        new Person("Артем", "Зимов", 45)
+    }, persons);
+  }
+
+  @Test
+  public void sortPersonsByLastNameThenFirstNameUsingArraysSortAnonymousComparator() {
+    Person[] persons = getPersons();
+
+    Arrays.sort(persons, new Comparator<Person>() {
+      @Override
+      public int compare(Person left, Person right) {
+        int LastNameCompare = left.getLastName().compareTo(right.getLastName());
+        return LastNameCompare == 0 ?
+            left.getFirstName().compareTo(right.getFirstName()) : LastNameCompare;
+      }
+    });
+
+    assertArrayEquals(new Person[]{
+        new Person("Алексей", "Доренко", 40),
+        new Person("Артем", "Зимов", 45),
+        new Person("Николай", "Зимов", 30),
+        new Person("Иван", "Мельников", 20)
+    }, persons);
+  }
+
+  @Test
+  public void findFirstWithAge30UsingGuavaPredicate() {
+    List<Person> persons = Arrays.asList(getPersons());
+
+    Predicate<Person> isAge30Checker = new Predicate<Person>() {
```

Comment:
Here must be declared class which implements Predicate and it's instance, not anonymous class

Reply:
Thanks.
Fixed.